[Flex] Rename FlexibleBoxAlgorithm to FlexLineBuilder

sammygill opened the pull request:
#### 9708c58850420957a977e733e8dd1c43b110765d
<pre>
[Flex] Rename FlexibleBoxAlgorithm to FlexLineBuilder
<a href="https://bugs.webkit.org/show_bug.cgi?id=292731">https://bugs.webkit.org/show_bug.cgi?id=292731</a>
<a href="https://rdar.apple.com/150944434">rdar://150944434</a>

Reviewed by Vitor Roriz and Alan Baradlay.

The name FlexibleBoxAlgorithm/FlexLayoutAlgorithm is very broad. The
flex layout algorithm as defined in the spec consists of various steps
such as computing flex item base sizes, creating flex lines, computing
cross sizes, etc. This class however simply builds up flex lines so
let&apos;s rename it to reflect that.

* Source/WebCore/SaferCPPExpectations/NoUncheckedPtrMemberCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncheckedLocalVarsCheckerExpectations:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/rendering/FlexLineBuilder.cpp: Renamed from Source/WebCore/rendering/FlexibleBoxAlgorithm.cpp.
(WebCore::FlexLayoutItem::FlexLayoutItem):
(WebCore::FlexLineBuilder::FlexLineBuilder):
(WebCore::FlexLineBuilder::canFitItemWithTrimmedMarginEnd const):
(WebCore::FlexLineBuilder::removeMarginEndFromFlexSizes const):
(WebCore::FlexLineBuilder::computeNextFlexLine):
(WebCore::FlexLayoutItem::constrainSizeByMinMax const):
* Source/WebCore/rendering/FlexLineBuilder.h: Renamed from Source/WebCore/rendering/FlexibleBoxAlgorithm.h.
(WebCore::FlexLayoutItem::hypotheticalMainAxisMarginBoxSize const):
(WebCore::FlexLayoutItem::flexBaseMarginBoxSize const):
(WebCore::FlexLayoutItem::flexedMarginBoxSize const):
(WebCore::FlexLayoutItem::style const):
(WebCore::FlexLineBuilder::isMultiline const):
* Source/WebCore/rendering/RenderFlexibleBox.cpp:
(WebCore::RenderFlexibleBox::performFlexLayout):
* Source/WebCore/rendering/RenderFlexibleBox.h:

Canonical link: <a href="https://commits.webkit.org/294685@main">https://commits.webkit.org/294685@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b3608ddaf61b241ec9c716d28ffd487a4ce649de

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/102666 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/22339 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/12657 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/107831 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/53307 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/104705 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/22680 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/30841 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/78089 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/35064 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/105672 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/17575 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/92653 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/58421 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/17416 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/10683 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/52664 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/87231 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/10754 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/110207 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/29803 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/21977 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/87069 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/30167 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/88844 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/86674 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/22068 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/31501 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/9231 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/24056 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/29730 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/35046 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/29538 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/32864 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/31100 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

vitorroriz reviewed (approved):
Not being very familiar with this part of the code and by reading your commit description I trust you that these names work better. Code-wise this LGTM.

alanbaradlay reviewed (approved):
Much better name, though I think it still implies a lot more than what this class actually does. We essentially just compute the range of flex items that belong to a flex line and provide some total values here. 'builder' normally does more than that...but this is certainly a good change.

sammygill committed:
[Flex] Rename FlexibleBoxAlgorithm to FlexLineBuilder
https://bugs.webkit.org/show_bug.cgi?id=292731
rdar://150944434

Reviewed by Vitor Roriz and Alan Baradlay.

The name FlexibleBoxAlgorithm/FlexLayoutAlgorithm is very broad. The
flex layout algorithm as defined in the spec consists of various steps
such as computing flex item base sizes, creating flex lines, computing
cross sizes, etc. This class however simply builds up flex lines so
let's rename it to reflect that.

* Source/WebCore/SaferCPPExpectations/NoUncheckedPtrMemberCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncheckedLocalVarsCheckerExpectations:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/rendering/FlexLineBuilder.cpp: Renamed from Source/WebCore/rendering/FlexibleBoxAlgorithm.cpp.
(WebCore::FlexLayoutItem::FlexLayoutItem):
(WebCore::FlexLineBuilder::FlexLineBuilder):
(WebCore::FlexLineBuilder::canFitItemWithTrimmedMarginEnd const):
(WebCore::FlexLineBuilder::removeMarginEndFromFlexSizes const):
(WebCore::FlexLineBuilder::computeNextFlexLine):
(WebCore::FlexLayoutItem::constrainSizeByMinMax const):
* Source/WebCore/rendering/FlexLineBuilder.h: Renamed from Source/WebCore/rendering/FlexibleBoxAlgorithm.h.
(WebCore::FlexLayoutItem::hypotheticalMainAxisMarginBoxSize const):
(WebCore::FlexLayoutItem::flexBaseMarginBoxSize const):
(WebCore::FlexLayoutItem::flexedMarginBoxSize const):
(WebCore::FlexLayoutItem::style const):
(WebCore::FlexLineBuilder::isMultiline const):
* Source/WebCore/rendering/RenderFlexibleBox.cpp:
(WebCore::RenderFlexibleBox::performFlexLayout):
* Source/WebCore/rendering/RenderFlexibleBox.h:

Canonical link: https://commits.webkit.org/294685@main
